### PR TITLE
reference elrond-go-storage

### PR DESCRIPTION
**Pull request by bogdan-rosianu**
## Description of the reasoning behind the pull request (what feature was missing / how the problem was manifesting itself / what was the motive behind the refactoring)
- the storers and cachers have been moved to a new repo `elrond-go-storage`
  
## Proposed Changes
- reference the new repo + remove the moved code from elrond-go 

## Testing procedure
- everything should work as before

### CODE REVIEW

File: factory/processComponents.go [hunk 1]
```diff
@@ -15,6 +15,9 @@ import (
 	dataBlock "github.com/ElrondNetwork/elrond-go-core/data/block"
 	"github.com/ElrondNetwork/elrond-go-core/data/indexer"
 	logger "github.com/ElrondNetwork/elrond-go-logger"
+	storageErrorsFactory "github.com/ElrondNetwork/elrond-go-storage/factory"
```
Comment (sstanculeanu) [inline factory/processComponents.go:18]:
```suggestion
	storageFactory "github.com/ElrondNetwork/elrond-go-storage/factory"
```

File: p2p/libp2p/metrics/printConnectionsWatcher.go [hunk 2]
```diff
@@ -9,8 +9,8 @@ import (
 	"github.com/ElrondNetwork/elrond-go-core/core"
 	"github.com/ElrondNetwork/elrond-go-core/core/atomic"
 	logger "github.com/ElrondNetwork/elrond-go-logger"
-	"github.com/ElrondNetwork/elrond-go/storage"
-	"github.com/ElrondNetwork/elrond-go/storage/timecache"
+	"github.com/ElrondNetwork/elrond-go-storage/timecache"
+	storage "github.com/ElrondNetwork/elrond-go-storage/types"
```
Comment (sstanculeanu) [inline p2p/libp2p/metrics/printConnectionsWatcher.go:13]: maybe
```suggestion
	storageTypes "github.com/ElrondNetwork/elrond-go-storage/types"
```

File: factory/processComponents.go (hunk 1 above)
Comment (bogdan-rosianu) [reply factory/processComponents.go:18]: `storageFactory` is already used. renamed to `storageRepoFactory`

File: p2p/libp2p/metrics/printConnectionsWatcher.go (hunk 2 above)
Comment (bogdan-rosianu) [reply p2p/libp2p/metrics/printConnectionsWatcher.go:13]: done

**Review by sstanculeanu — APPROVED** [review]

File: go.mod [hunk 3]
```diff
@@ -6,12 +6,12 @@ require (
 	github.com/ElrondNetwork/arwen-wasm-vm/v1_2 v1.2.42-0.20220729115258-b9f2fb2f6568
 	github.com/ElrondNetwork/arwen-wasm-vm/v1_3 v1.3.42-0.20220729115131-85ecca868e90
 	github.com/ElrondNetwork/arwen-wasm-vm/v1_4 v1.4.59-0.20220729115431-a6c93119bdda
-	github.com/ElrondNetwork/concurrent-map v0.1.3
 	github.com/ElrondNetwork/covalent-indexer-go v1.0.6
 	github.com/ElrondNetwork/elastic-indexer-go v1.2.38
 	github.com/ElrondNetwork/elrond-go-core v1.1.18
 	github.com/ElrondNetwork/elrond-go-crypto v1.0.1
 	github.com/ElrondNetwork/elrond-go-logger v1.0.7
+	github.com/ElrondNetwork/elrond-go-storage v0.0.0-20220811102708-2dc21ec20863
```
Comment (AdoAdoAdo) [inline go.mod:14]: don't forget to replace with a tag before merge

**Review by AdoAdoAdo — DISMISSED** [review]

File: go.mod (hunk 3 above)
Comment (bogdan-rosianu) [reply go.mod]: I have added a tag for `elrond-go-storage`. For arwen, I think they should come in a new PR

**Review by sstanculeanu — APPROVED** [review]

**Review by AdoAdoAdo — APPROVED** [review]

**Review by sstanculeanu — APPROVED** [review]

**Review by gabi-vuls — APPROVED** [review]
System test passed.

**Pull request merged**